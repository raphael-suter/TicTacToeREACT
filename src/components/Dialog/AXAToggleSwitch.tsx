import createAXAToggleSwitchReact from "@axa-ch/toggle-switch/lib/index.react";
import { createElement } from "react";

const AXAToggleSwitch = createAXAToggleSwitchReact(createElement);

export default AXAToggleSwitch;
