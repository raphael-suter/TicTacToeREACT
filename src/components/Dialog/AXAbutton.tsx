import createAXAButtonReact from "@axa-ch/button/lib/index.react";
import { createElement } from "react";
import styled from "styled-components";

const AXAbutton = styled(createAXAButtonReact(createElement, "ttt"))`
  margin-top: 0.6rem;
`;

export default AXAbutton;
