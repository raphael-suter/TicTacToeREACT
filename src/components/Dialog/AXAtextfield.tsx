import createAXAInputTextReact from '@axa-ch/input-text/lib/index.react';
import { createElement } from 'react';
import styled from 'styled-components';

const AXAtextfield = styled(createAXAInputTextReact(createElement))`
  display: block;
  margin-bottom: 0.6rem;
`;

export default AXAtextfield;
