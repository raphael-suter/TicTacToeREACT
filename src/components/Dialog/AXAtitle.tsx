import createAXAHeadingReact from '@axa-ch/heading/lib/index.react';
import { createElement } from 'react';

export default createAXAHeadingReact(createElement);
