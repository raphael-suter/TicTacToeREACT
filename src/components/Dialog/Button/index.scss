@import "../../../styles/colors.scss";

.button {
  margin: 0.6rem;
  padding: 0.5rem 2rem;
  text-transform: uppercase;
  font-size: 16px;
  border: none;
  background: $mainColor;
  color: $lightColor;
  cursor: pointer;
}
