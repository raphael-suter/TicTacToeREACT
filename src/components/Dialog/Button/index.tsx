import React from 'react';
import './index.scss';
import Props from './Props';

export default ({ text, onClick }: Props) => (
  <button className='dialog-button' onClick={onClick}>{text}</button>
);
