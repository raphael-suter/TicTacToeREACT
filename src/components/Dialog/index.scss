@import "../../styles/colors.scss";
@import "../../styles/mixins.scss";

.dialog {
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 1.2rem;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: $shadow;

  form {
    @include card;
    flex: 1;
  }
}
