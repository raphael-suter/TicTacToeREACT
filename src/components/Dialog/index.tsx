import React from 'react';
import './index.scss';

interface Props {
  visible: boolean;
  children: JSX.Element[];
}

const Dialog = ({ visible, children }: Props) => (
  <div className='dialog' style={{ visibility: visible ? 'visible' : 'hidden' }}>
    <form>
      {children}
    </form>
  </div>
);

export { TextField } from './TextField';
export { Button } from './Button';
export default Dialog;
