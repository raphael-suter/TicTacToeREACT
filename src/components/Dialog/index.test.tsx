import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import Dialog from '.';

describe('Dialog', () => {
  test('Renders correctly.', () => {
    const wrapper = shallow(<Dialog children={null} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
