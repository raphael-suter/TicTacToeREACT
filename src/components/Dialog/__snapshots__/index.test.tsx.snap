// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog Renders correctly. 1`] = `
<styled.div>
  <styled.article />
</styled.div>
`;
